allennlp/models/encoder_decoders/nlvr_semantic_parser.py

```diff
@@ -0,0 +1,365 @@
+                                               actions_to_entities=None,
+                                               entity_types=None)
+        self.agenda = agenda
+        self.agenda_mask = agenda_mask
+        self.checklist = checklist
+
+    @classmethod
+    def combine_states(cls, states) -> 'NlvrDecoderState':
+        super_state = super(NlvrDecoderState, cls).combine_states(states)
+        agenda = [alist for state in states for alist in state.agenda]
+        agenda_mask = [mask_list for state in states for mask_list in state.agenda_mask]
+        checklist = [clist for state in states for clist in state.checklist]
+        return NlvrDecoderState(agenda,
+                                agenda_mask,
+                                checklist,
+                                super_state.batch_indices,
+                                super_state.action_history,
+                                super_state.score,
+                                super_state.hidden_state,
+                                super_state.memory_cell,
+                                super_state.previous_action_embedding,
+                                super_state.attended_question,
+                                super_state.grammar_state,
+                                states[0].encoder_outputs,
+                                states[0].encoder_output_mask,
+                                states[0].action_embeddings,
+                                states[0].action_indices,
+                                states[0].possible_actions)
+
+    @overrides
+    def _make_new_state_with_group_indices(self, group_indices: List[int]) -> 'NlvrDecoderState':
+        super_state = super(NlvrDecoderState, self)._make_new_state_with_group_indices(group_indices)
+        group_agenda = [self.agenda[i] for i in group_indices]
+        group_agenda_mask = [self.agenda_mask[i] for i in group_indices]
+        group_checklist = [self.checklist[i] for i in group_indices]
+        return NlvrDecoderState(group_agenda,
+                                group_agenda_mask,
+                                group_checklist,
+                                super_state.batch_indices,
+                                super_state.action_history,
+                                super_state.score,
+                                super_state.hidden_state,
+                                super_state.memory_cell,
+                                super_state.previous_action_embedding,
+                                super_state.attended_question,
+                                super_state.grammar_state,
+                                self.encoder_outputs,
+                                self.encoder_output_mask,
+                                self.action_embeddings,
+                                self.action_indices,
+                                self.possible_actions)
+
+
+class NlvrDecoderStep(WikiTablesDecoderStep):
```

Comment:
Don't inherit from the wikitables model, copy what you need.

Reply:
Done.